roborazzi-compose-preview-scanner-support/src/main/java/com/github/takahirom/roborazzi/RoborazziPreviewScannerSupport.kt

```diff
@@ -16,33 +18,83 @@ fun ComposablePreview<AndroidPreviewInfo>.captureRoboImage(
     composablePreview()
   }
 }
-
+/**
+ * ComposePreviewTester is an interface that allows you to define a custom test for a Composable preview.
+ * The class that implements this interface should have a parameterless constructor.
+ * You can set the custom tester class name in the roborazzi.generateComposePreviewRobolectricTests.testerQualifiedClassName property.
+ * A new instance of the tester class is created for each test execution and preview generation.
+ */
 @ExperimentalRoborazziApi
 interface ComposePreviewTester<T : Any> {
+  data class Options(
+    val testLifecycleOptions: TestLifecycleOptions = JUnit4TestLifecycleOptions(),
+    val scanOptions: ScanOptions = ScanOptions(emptyList()),
+  ) {
+    interface TestLifecycleOptions
+    data class JUnit4TestLifecycleOptions(
+      /**
+       * The TestRule factory to be used for the generated tests.
+       * You can use this to add custom behavior to the generated tests.
+       */
+      // Used from generated tests
+      @Suppress("unused") val testRuleFactory: () -> TestRule = { object : TestWatcher() {} },
+    ) : TestLifecycleOptions
+
+    data class ScanOptions(
+      /**
+       * The packages to scan for composable previews.
+       */
+      val packages: List<String>,
+      /**
+       * Whether to include private previews in the scan.
+       */
+      val includePrivatePreviews: Boolean = false,
+    )
+  }
+
+  /**
+   * Retrieves the options for the ComposePreviewTester.
+   *
+   * @return The Options object.
+   */
+  fun options(): Options = defaultOptionsFromPlugin
+
   /**
    * Retrieves a list of composable previews from the specified packages.
    *
    * @param packages Vararg parameter representing the package names to scan for previews.
    * @return A list of ComposablePreview objects of type T.
    */
-  fun previews(vararg packages: String): List<ComposablePreview<T>>
+  fun previews(): List<ComposablePreview<T>>
```

Comment:
This is going to be a breaking change. I considered providing a default method, but I think it would make it difficult for users to know what to do.

Reply:
The release note will be like this.

## Breaking changes for users of the `ComposePreviewTester` interface

We've released [Compose Preview Support](https://takahirom.github.io/roborazzi/preview-support.html) in version 1.22.0. You can now modify the behavior of Compose Preview Support using ComposePreviewTester. Previously, the API of ComposePreviewTester was prone to breaking changes with each new option added. We've addressed this issue by introducing a new `options()` function. However, this introduces a breaking change for current users of ComposePreviewTester.

Old interface:
```kotlin
fun previews(vararg packages: String): List<ComposablePreview<T>>
```

New interface (Now you can access packages via `options().scanOptions.packages`):
```kotlin
fun previews(): List<ComposablePreview<T>>
```

## New feature: Support for includePrivatePreviews in Compose Preview Support

The `includePrivatePreviews` option allows you to include private previews in Compose Preview Support. You can set `includePrivatePreviews` in `roborazzi.generateComposePreviewRobolectricTests.includePrivatePreviews`.

## New feature: JUnit rule support in ComposePreviewTester

Previously, ComposePreviewTester lacked lifecycle hooks, making it challenging to handle certain scenarios. Now, you can pass your own Test rules, including your Compose Test Rule, and use them in tests. For a sample implementation, refer to [this integration test](https://github.com/takahirom/roborazzi/blob/2519f46ec0e5a8dadc5139ae833e2925c90c8b21/include-build/roborazzi-gradle-plugin/src/integrationTest/projects/sample-generate-preview-tests/src/test/java/com/github/takahirom/sample/CustomPreviewTester.kt).